# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/DirectionalityUtils.cpp
# Commit: 4f30d7afdf22
# Full Hash: 4f30d7afdf22c652481513564b60b37053c0948b
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Fix build bustage" for causing build bustages.
#   
#   This reverts commit 0681b181d6a8a48ac460a97dacb7fca91f9989c3.
#   
#   Revert "Bug 1932150 - Fix lint failure" for causing Build bustage.
# ==============================================================================

diff -r 1e64a48e3517 -r 4f30d7afdf22 dom/base/DirectionalityUtils.cpp
--- a/dom/base/DirectionalityUtils.cpp	Wed May 07 20:11:46 2025 +0300
+++ b/dom/base/DirectionalityUtils.cpp	Wed May 07 20:37:37 2025 +0300
@@ -559,19 +559,6 @@
 }
 
 void SlotAssignedNodeAdded(HTMLSlotElement* aSlot, nsIContent& aAssignedNode) {
-  MOZ_ASSERT(aSlot);
-  if (StaticPrefs::dom_shadowdom_selection_across_boundary_enabled()) {
-    if (aSlot->IsMaybeSelected()) {
-      // Normally it's nsRange::ContentAppended's responsibility to
-      // mark new descendants, however this doesn't work for slotted
-      // content because nsRange observes the common ancestor of
-      // start/end, whereas slotted element may not have the same
-      // ancestor as them.
-      dom::AbstractRange::UpdateDescendantsInFlattenedTree(
-          aAssignedNode, true /* aMarkDesendants*/);
-    }
-  }
-
   if (aSlot->HasDirAuto()) {
     aAssignedNode.SetAffectsDirAutoSlot();
     DownwardPropagateDirAutoFlags(&aAssignedNode);
@@ -581,19 +568,6 @@
 
 void SlotAssignedNodeRemoved(HTMLSlotElement* aSlot,
                              nsIContent& aUnassignedNode) {
-  if (StaticPrefs::dom_shadowdom_selection_across_boundary_enabled() &&
-      aUnassignedNode.IsMaybeSelected()) {
-    // Normally, this shouldn't happen because nsRange::ContentRemoved
-    // should be called for content removal, and then
-    // AbstractRange::UnmarkDescendants will be used to clear the flags.
-    // Though this doesn't work for slotted element because nsRange
-    // observers the common ancestor of start/end, whereas slotted element
-    // may not have the same ancestor as them, so we have to clear
-    // the flags manually here.
-    dom::AbstractRange::UpdateDescendantsInFlattenedTree(
-        aUnassignedNode, false /* aMarkDesendants*/);
-  }
-
   if (aSlot->HasDirAuto()) {
     MaybeClearAffectsDirAutoSlot(&aUnassignedNode);
   }